nixos/modules/config/nix.nix

```diff
@@ -386,5 +386,11 @@ in
         )
       );
     };
+
+    # Enable git if flake support is enabled
+    programs.git = mkIf (builtins.match ".*flakes.*" != null) {
```

Comment:
Instead of a regex match, wouldn't it be better to check if `nixpkgs.flake.source` isn't null? This is already done with [nixpkgs.flake.setNixPath](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/misc/nixpkgs-flake.nix#L37-L55) and [nixpkgs.flake.setFlakeRegistry](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/misc/nixpkgs-flake.nix#L57-L76), which are enabled by default if the NixOS system was defined using Flakes.